eureka/S5_lightcurve_fitting/models/BatmanModels.py

```diff
@@ -117,6 +128,12 @@ def eval(self, **kwargs):
             for index,item in enumerate(self.longparamlist[c]):
                 setattr(bm_params, self.paramtitles[index], self.parameters.dict[item][0])
 
+            # Enforce physicality to avoid crashes from batman by returning something that should be a horrible fit
+            if not ((0 < bm_params.fp < 1) and (0 < bm_params.rp < 1) and (0 < bm_params.per) and (0 < bm_params.inc < 90) and (1 < bm_params.a) and (0 <= bm_params.ecc < 1) and (0 <= bm_params.w <= 360)):
```

Comment:
Well it's complicated - we do know for a fact that Fp/Fs must be strictly greater than zero, so having a prior that allows for negative values could be thought of as biasing you towards lower values since you're claiming that negative values are just as probable as positive values. Ideally one would just quote an upper limit if their value is consistent with zero to avoid any perceived bias from enforcing Fp/Fs>0, but that doesn't always happen in practice. I'll agree that this is contentious enough to not be baked directly into the code though and I'll remove it. I had added it as I wanted to use broad Gaussian priors for my fits for the ERS data challange but also wanted to enforce physical eclipse depths. Perhaps we can discuss the possibility of a truncated normal prior and other more complicated priors in the future.

Reply:
We should discuss this next week, in person. I'll explain why I think it's a bad idea